fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -2173,51 +2177,71 @@ void setLakeStorageInfo(OlapTable table, String storageVolumeId, Map<String, Str
         table.setStorageInfo(pathInfo, dataCacheInfo);
     }
 
-    void onCreate(Database db, Table table, String storageVolumeId, boolean isSetIfNotExists) throws DdlException {
+    public void onCreate(Database db, Table table, String storageVolumeId, boolean isSetIfNotExists)
+            throws DdlException {
         // check database exists again, because database can be dropped when creating table
         if (!tryLock(false)) {
-            throw new DdlException("Failed to acquire globalStateMgr lock. Try again");
+            throw new DdlException("Failed to acquire globalStateMgr lock. " +
+                    "Try again or increasing value of `catalog_try_lock_timeout_ms` configuration.");
         }
 
         try {
+            /*
+             * When creating table or mv, we need to create the tablets and prepare some of the
+             * metadata first before putting this new table or mv in the database. So after the
+             * first step, we need to acquire the global lock and double check whether the db still
+             * exists because it maybe dropped by other concurrent client. And if we don't use the lock
+             * protection and handle the concurrency properly, the replay of table/mv creation may fail
+             * on restart or on follower.
+             *
+             * After acquire the db lock, we also need to acquire the db lock and write edit log. Since the
+             * db lock maybe under high contention and IO is busy, current thread can hold the global lock
+             * for quite a long time and make the other operation waiting for the global lock fail.
+             *
+             * So here after the confirmation of existence of modifying database, we release the global lock
+             * When dropping database, we will set the `exist` field of db object to false. And in the following
+             * creation process, we will double-check the `exist` field.
+             */
             if (getDb(db.getId()) == null) {
-                throw new DdlException("Database has been dropped when creating table");
+                throw new DdlException("Database has been dropped when creating table/mv/view");
             }
+        } finally {
+            unlock();
+        }
 
-            if (db.isSystemDatabase()) {
-                ErrorReport.reportDdlException(ErrorCode.ERR_CANT_CREATE_TABLE, table.getName(),
-                        "cannot create table in system database");
-            }
+        if (db.isSystemDatabase()) {
+            ErrorReport.reportDdlException(ErrorCode.ERR_CANT_CREATE_TABLE, table.getName(),
```

Comment:
Add the database name in the exception

Reply:
it's the system database, already reported...